PostgreSQL: dynamic `Sort` does not quote explicitly mapped uppercase column names

### Expected Behavior

Micronaut Data should generate valid PostgreSQL SQL and consistently quote the explicitly mapped uppercase identifier in every clause, including ORDER BY:

  ORDER BY owner_."LAST_NAME" ASC

  The query should return the two matching owners.

### Actual Behaviour

 Micronaut Data generates SQL that correctly quotes the mapped column in the SELECT and WHERE clauses but does not quote it in ORDER BY:

  SELECT owner_."id",owner_."FIRST_NAME",owner_."LAST_NAME",
         owner_."ADDRESS",owner_."CITY",owner_."TELEPHONE"
  FROM "OWNERS" owner_
  WHERE (owner_."LAST_NAME" ILIKE CONCAT('%',?,'%'))
  ORDER BY owner_.LAST_NAME ASC

  PostgreSQL folds the unquoted LAST_NAME to lowercase and fails because the actual column is the case-sensitive "LAST_NAME":

  ERROR: column owner_.last_name does not exist
  Position: 191

  Stacktrace excerpt:

  io.micronaut.data.exceptions.DataAccessException: Error executing SQL Query:
  ... ORDER BY owner_.LAST_NAME ASC
  ERROR: column owner_.last_name does not exist

  Caused by: org.postgresql.util.PSQLException:
  ERROR: column owner_.last_name does not exist
  Position: 191

### Steps To Reproduce

 1. Use an entity with an explicitly uppercase mapped column:

     ```java
     @MappedProperty("LAST_NAME")
     String lastName

  2. Define a PostgreSQL repository:

     @JdbcRepository(dialect = Dialect.POSTGRES)
     interface PostgresOwnerRepository extends OwnerRepository {
     }

  3. Invoke a derived finder with a dynamic sort:

     ownerRepository.findByLastNameContainingIgnoreCase(
         "davis",
         Sort.of(Sort.Order.asc("lastName"))
     );

  4. Run the reproducer with the PostgreSQL environment enabled:

     MICRONAUT_ENVIRONMENTS=postgres ./gradlew test

### Environment Information

- Operating System: macOS 26.5.2
- Micronaut Version: 4.10.17
- Micronaut Data JDBC Version: 4.14.4
- JDK Version: Oracle GraalVM JDK 25.0.2
- Database: PostgreSQL via Testcontainers (postgres:latest)

### Example Application

https://github.com/ZARAYACH/jdbcRepoPostgresIssue.git

### Version

4.10.17

--- a/data-jdbc/src/test/groovy/io/micronaut/data/jdbc/postgres/PostgresRepositorySpec.groovy
+++ b/data-jdbc/src/test/groovy/io/micronaut/data/jdbc/postgres/PostgresRepositorySpec.groovy
@@ -19,6 +19,7 @@ import groovy.transform.Memoized
 import io.micronaut.data.model.Sort
 import io.micronaut.data.tck.entities.AssignedIdReturningEntity
 import io.micronaut.data.tck.entities.Book
+import io.micronaut.data.tck.entities.Owner
 import io.micronaut.data.tck.repositories.*
 import io.micronaut.data.tck.tests.AbstractRepositorySpec
 
@@ -185,6 +186,11 @@ class PostgresRepositorySpec extends AbstractRepositorySpec implements PostgresT
     @Override
     IntervalRepository getIntervalRepository() {
         return context.getBean(PostgresIntervalRepository)
+    }
+
+    @Memoized
+    PostgresOwnerRepository getOwnerRepository() {
+        return context.getBean(PostgresOwnerRepository)
     }
 
     @Memoized
@@ -743,4 +749,30 @@ class PostgresRepositorySpec extends AbstractRepositorySpec implements PostgresT
         result2.tags() == ["foo", "bar"]
     }
 
+    private Owner createOwner(String name) {
+        Owner owner = new Owner(name)
+        return ownerRepository.save(owner)
+    }
+
+    void 'test uppercase column names'() {
+        given:
+        createOwner("George Franklin")
+        createOwner("Betty Davis")
+        createOwner("Eduardo Rodriquez")
+        createOwner("Harold Davis")
+        createOwner("Peter McTavish")
+        createOwner("Jean Coleman")
+        createOwner("Jeff Black")
+        createOwner("Maria Escobito")
+        when:
+        def owners = ownerRepository.findByNameContainingIgnoreCase("davis", Sort.of(Sort.Order.asc("name")))
+        then:
+        owners.size() == 2
+        when:
+        owners = ownerRepository.findByNameNativeQuery("Eduardo Rodriquez")
+        then:
+        owners.size() == 1
+        cleanup:
+        ownerRepository.deleteAll()
+    }
 }

--- a/data-jdbc/src/test/java/io/micronaut/data/jdbc/postgres/PostgresOwnerRepository.java
+++ b/data-jdbc/src/test/java/io/micronaut/data/jdbc/postgres/PostgresOwnerRepository.java
@@ -1,0 +1,30 @@
+package io.micronaut.data.jdbc.postgres;
+
+import io.micronaut.data.annotation.Query;
+import io.micronaut.data.jdbc.annotation.JdbcRepository;
+import io.micronaut.data.model.Sort;
+import io.micronaut.data.model.query.builder.sql.Dialect;
+import io.micronaut.data.repository.CrudRepository;
+import io.micronaut.data.tck.entities.Owner;
+
+import java.util.Collection;
+import java.util.List;
+
+@JdbcRepository(dialect = Dialect.POSTGRES)
+public interface PostgresOwnerRepository extends CrudRepository<Owner, Integer> {
+
+    /**
+     * Finds owners by case-insensitive partial name match.
+     *
+     * @param name the name fragment to search for
+     * @param sort the sort order for results
+     * @return matching owners
+     */
+    Collection<Owner> findByNameContainingIgnoreCase(String name, Sort sort);
+
+    @Query(value = """
+                SELECT o.* FROM "OWNERS" o where o."OWNER_NAME"=:name
+                """, nativeQuery = true)
+    List<Owner> findByNameNativeQuery(String name);
+
+}

--- a/data-model/src/main/java/io/micronaut/data/model/query/builder/sql/AbstractSqlLikeQueryBuilder.java
+++ b/data-model/src/main/java/io/micronaut/data/model/query/builder/sql/AbstractSqlLikeQueryBuilder.java
@@ -1285,7 +1285,9 @@ public abstract class AbstractSqlLikeQueryBuilder implements QueryBuilder {
                 appendJsonProjection(buff, path.getProperty().getDataType());
             }
         } else {
-            buff.append(getColumnName(path.getProperty()));
+            String columnName = getColumnName(path.getProperty());
+            PersistentEntity propertyOwner = path.findPropertyOwner().orElse(path.getProperty().getOwner());
+            buff.append(escapeColumnIfNeeded(columnName, shouldEscape(propertyOwner)));
         }
 
         return buff.toString();

--- a/data-processor/src/test/groovy/io/micronaut/data/model/query/builder/SqlQueryBuilderSpec.groovy
+++ b/data-processor/src/test/groovy/io/micronaut/data/model/query/builder/SqlQueryBuilderSpec.groovy
@@ -42,6 +42,7 @@ import io.micronaut.data.tck.entities.Book
 import io.micronaut.data.tck.entities.Car
 import io.micronaut.data.tck.entities.City
 import io.micronaut.data.tck.entities.CountryRegion
+import io.micronaut.data.tck.entities.Owner
 import io.micronaut.data.tck.entities.Product
 import io.micronaut.data.tck.entities.Restaurant
 import io.micronaut.data.tck.entities.Sale
@@ -461,10 +462,22 @@ interface MyRepository {
 
         where:
         type   | direction | props              | statement
-        Person | 'asc'     | ["name"]           | 'person_.name ASC'
-        Person | 'asc'     | ["name", "someId"] | 'person_.name ASC,person_.some_id ASC'
-        Person | 'desc'    | ["name"]           | 'person_.name DESC'
-        Person | 'desc'    | ["name", "someId"] | 'person_.name DESC,person_.some_id DESC'
+        Person | 'asc'     | ["name"]           | 'person_.`name` ASC'
+        Person | 'asc'     | ["name", "someId"] | 'person_.`name` ASC,person_.`some_id` ASC'
+        Person | 'desc'    | ["name"]           | 'person_.`name` DESC'
+        Person | 'desc'    | ["name", "someId"] | 'person_.`name` DESC,person_.`some_id` DESC'
+    }
+
+    void 'test encode order by uppercase mapped column'() {
+        given:
+        PersistentEntity entity = new RuntimePersistentEntity(Owner)
+        Sort sort = Sort.of(Sort.Order.asc('name'))
+
+        when:
+        String query = new SqlQueryBuilder(Dialect.POSTGRES).buildOrderBy('', entity, AnnotationMetadata.EMPTY_METADATA, sort, false, null)
+
+        then:
+        query == ' ORDER BY owner_."OWNER_NAME" ASC'
     }
 
     void "test encode insert statement"() {

--- a/data-tck/src/main/java/io/micronaut/data/tck/entities/Owner.java
+++ b/data-tck/src/main/java/io/micronaut/data/tck/entities/Owner.java
@@ -19,13 +19,15 @@ import io.micronaut.core.annotation.Creator;
 import io.micronaut.data.annotation.GeneratedValue;
 import io.micronaut.data.annotation.Id;
 import io.micronaut.data.annotation.MappedEntity;
+import io.micronaut.data.annotation.MappedProperty;
 
-@MappedEntity
+@MappedEntity("OWNERS")
 public class Owner {
 
     @Id
     @GeneratedValue
     private Long id;
+    @MappedProperty("OWNER_NAME")
     private String name;
     private int age;
 

--- a/gradle/libs.versions.toml
+++ b/gradle/libs.versions.toml
@@ -1,5 +1,5 @@
 [versions]
-micronaut = "5.1.6"
+micronaut = "5.1.10"
 micronaut-platform = "5.0.4"
 micronaut-docs = "3.0.0"
 micronaut-gradle-plugin = "5.0.2"

